dev/sparktestsupport/modules.py

```diff
@@ -19,10 +19,28 @@
 import itertools
 import os
 import re
+import glob
+
+from sparktestsupport import SPARK_HOME
 
 all_modules = []
 
 
+def _discover_python_unittests(paths):
+    if not paths:
+        return set([])
+    tests = set([])
+    pyspark_path = os.path.join(SPARK_HOME, "python")
+    for path in paths:
+        # Discover the test*.py in every path
+        files = glob.glob(os.path.join(pyspark_path, path, "test_*.py"))
+        for f in files:
+            # Convert 'pyspark_path/pyspark/tests/test_abc.py' to 'pyspark.tests.test_abc'
+            file2module = os.path.relpath(f, pyspark_path)[:-3].replace("/", ".")
```

Comment:
Hmm .. I like the idea but the implementation is slightly flaky ... e.g.) I wonder if we can import pyspark and go through the sub packages manually instead of going through the files ..

Reply:
or .. define a private variable in each file, and decide which tests are slow, etc. based on that .. ?

Reply:
> I wonder if we can import pyspark and go through the sub packages manually instead of going through the files ..

Yep, it's more accurate, I change the rules from "test_* discover" to "unittest module discover", it will only discover the modules which are interit from unittest.TestCase. See my latest change: https://github.com/apache/spark/pull/32867/commits/eb835c5d73b4f844cdd6dff1fd37a2317079c5e6